providers/summary/scancode.js

```diff
@@ -84,10 +91,26 @@ class ScanCodeSummarizer {
     if (!licenses) return null
     const list = setToArray(licenses)
     if (!list) return null
-    return list
-      .map(normalizeSpdx)
-      .filter(x => x)
-      .join(' AND ')
+    if (list.length === 1) return list[0]
+    return list.map(x => `(${x})`).join(' AND ')
+  }
+
+  _createExpression(rule, licenseKey) {
+    const license_expression = get(rule, 'license_expression')
+    if (!license_expression) return normalizeSpdx(licenseKey)
+    let result = license_expression
+    rule.licenses.forEach(licence => {
```

Comment:
Suggest adding expression handling to `normalize` and then calling that.

Reply:
so this is actually very specific to the scancode output and is not general expression support

we get an array of licenses and an all-lowercase expression

```
"matched_rule": {
          "identifier": "mit_or_apache-2.0_2.RULE",
          "license_expression": "mit OR apache-2.0",
          "licenses": ["mit", "apache-2.0"],
          "matcher": "2-aho",
          "rule_length": 48,
          "matched_length": 48,
          "match_coverage": 100,
          "rule_relevance": 100
        }
```

we go over each license and normalize and replace it in the expression so that it will be parseable by spdx-expression-parse

Reply:
just fixing the casing of this licenses array may not be enough. Some of the scancode keys differ from spdx ids by more than just casing. Better to have a lookup from scancode key to spdx identifier